src/providers/postgres/qgspostgresprovider.h

```diff
@@ -485,6 +486,9 @@ class QgsPostgresProvider final: public QgsVectorDataProvider
     static int crsToSrid( const QgsCoordinateReferenceSystem &crs,  QgsPostgresConn *conn );
     static QgsCoordinateReferenceSystem sridToCrs( int srsId, QgsPostgresConn *conn );
 
+    bool estimateExtent() const;
+    bool computeExtent3D() const;
```

Comment:
Can we have a comment on theses functions as the code is becoming more and more complex? Just to remember why you have done this fix ;)

Reply:
I've pushed comments